lib/public_activity/utility/store_controller.rb

```diff
@@ -1,24 +1,12 @@
 module PublicActivity
-  # @private
-  @@controllers = Hash.new
-  # Lambda called after the thread is destroyed.
-  Finalizer = lambda { |id|
-    @@controllers.delete id
-  }
-
-  class << self
-    # Setter for remembering controller instance
-    def set_controller(controller)
-      unless @@controllers.has_key?(Thread.current.object_id)
-        ObjectSpace.define_finalizer Thread.current, Finalizer
-      end if RUBY_VERSION != "1.9.3"
-      @@controllers[Thread.current.object_id] = controller
-    end
+  # Setter for remembering controller instance
+  def self.set_controller(controller)
+    Thread.current[:public_activity_controller] = controller
+  end
 
-    # Getter for accessing the controller instance
-    def get_controller
-      @@controllers[Thread.current.object_id]
-    end
+  # Getter for accessing the controller instance
+  def self.get_controller
+    Thread.current[:public_activity_controller]
```

Comment:
Yes, the behavior doesn't change. The new way makes sure the variable gets cleaned up with the thread. We used to do ObjectSpace finalizers with the hash, but it's deprecated and not nearly as expressive.